compiler-core/src/type_/environment.rs

```diff
@@ -591,6 +591,8 @@ impl Environment<'_> {
                                 // Check this in the hydrator, i.e. is it a created type
                                 let v = self.new_unbound_var();
                                 let _ = ids.insert(*id, v.clone());
+                                let new_id = self.previous_uid();
+                                self.names.map_new_variable(*id, new_id);
```

Comment:
This looks good. What was the situation in which it wouldn't work without this change? A comment explaining it would be helpful for future readers.

Reply:
Absolutely, great feedback!

I added a comment explaining why this is necessary, also changed `new_id` name which was confusing, coming from a previous id... 🤔 

I hope it is more clear now!